opal/mca/accelerator/base/accelerator_base_select.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2004-2010 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2007 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2012-2015 Los Alamos National Security, LLC.  All rights
+ *                         reserved.
+ * Copyright (c) 2013-2020 Intel, Inc.  All rights reserved.
+ * Copyright (c) 2015-2020 Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2020-2022 Amazon.com, Inc. or its affiliates.  All Rights
+ * Copyright (c) 2018-2020 Triad National Security, LLC. All rights
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "opal_config.h"
+#include "opal/class/opal_list.h"
+#include "opal/util/output.h"
+#include "opal/util/show_help.h"
+#include "opal/util/proc.h"
+#include "opal/constants.h"
+#include "opal/mca/base/base.h"
+#include "opal/mca/mca.h"
+#include "opal/mca/accelerator/base/base.h"
+#include "opal/mca/accelerator/accelerator.h"
+
+typedef struct accelerator_opened_component_t {
+  opal_list_item_t super;
+  opal_accelerator_base_component_t *accelerator_component;
+} accelerator_opened_component_t;
+
+
+int opal_accelerator_base_select(void)
```

Comment:
I question whether this is the right selection logic.  We're saying that there's a "best" accelerator, which feels like a path to madness today.  I think there are two valid options for the selection logic:

1. We can have at most 1 active accelerator framework.  If we find multiple that say they work, we give up and make the user pick.  The at most 1 is important; 0 feels like a fairly normal case.
2. We can support multiple simultaneous accelerator components.